addons/base_automation/models/base_automation.py

```diff
@@ -1072,18 +1072,17 @@ def _cron_process_time_based_actions(self, *, auto_commit=True):
             now = self.env.cr.now()
             records = automation._search_time_based_automation_records(until=now)
             # run the automation on the records
-            try:
-                for record in records:
+            for record in records:
+                try:
                     automation._process(record)
-                self.env.flush_all()
-            except Exception as e:
-                if not auto_commit:
-                    raise
-                self.env.cr.rollback()
-                _logger.exception("Error in time-based automation rule `%s`.", automation.name)
-                final_exception = e
-                continue
-
+                except Exception as e:
+                    if not auto_commit:
+                        raise
+                    self.env.cr.rollback()
```

Comment:
We cannot commit/rollback here, the whole automation has to run in the same transaction.